Firmware/messages.c

```diff
@@ -18,36 +18,44 @@
 
 #include "base.h"
 
-void bp_message_write_buffer(unsigned long strptr, size_t length) {
-    size_t index;
+void bp_message_write_buffer(unsigned long strptr) {
     unsigned char tblpag_prev = TBLPAG;
+    unsigned char index = 0;
```

Comment:
Can you please use data types from `stdint.h`?  I'm (slowly) getting rid of usages of non-size-specified types to prevent issues when changing code, as it's easy to forget that on this platform `int` is 16 bits and not 32 (I'm guilty of that as well :))